### PR TITLE
FS-117 FS-119 Order-System but interactive

### PR DESCRIPTION
**Pull request by Jakllp**

### CODE REVIEW

File: addons/godot-xr-tools/objects/pickable.gd [hunk 1]
```diff
@@ -226,7 +226,8 @@ func pick_up(by: Node3D) -> void:
 	# Skip if not enabled
 	if not enabled:
 		return
-
+	
+	freeze = false
```
Comment (Thyrexx) [inline addons/godot-xr-tools/objects/pickable.gd:230]: why?

File: resources/food/tomato.tscn [hunk 2]
```diff
@@ -13,6 +13,8 @@ collision_mask = 8519687
 mass = 0.2
 center_of_mass_mode = 1
 center_of_mass = Vector3(0, 0.06, 0)
+freeze = true
```
Comment (Thyrexx) [inline resources/food/tomato.tscn]: why specifically the potato and tomato. Isn't that something that might apply to every food item?

File: resources/food/tomato.tscn [hunk 3]
```diff
@@ -13,6 +13,8 @@ collision_mask = 8519687
...
+continuous_cd = true
```
Comment (Thyrexx) [inline resources/food/tomato.tscn]: Same as above (?)

File: resources/food/potato.tscn [hunk 4]
```diff
@@ -11,6 +11,7 @@ height = 0.13169
 
 [node name="Potato" instance=ExtResource("1_ptydu")]
 mass = 0.25
+freeze = true
```
Comment (Thyrexx) [inline resources/food/potato.tscn]: See tomato.tscn

File: resources/objects/accessoires/tray.tscn [hunk 5]
```diff
@@ -21,6 +23,10 @@ radius = 0.04
 
 [node name="Tray" instance=ExtResource("1_xu6s3")]
 collision_mask = 8388615
+mass = 0.25
+freeze = true
+continuous_cd = true
```
Comment (Thyrexx) [inline resources/objects/accessoires/tray.tscn:28]: Is this strictly neccessray?
Same question for other accessoires.

File: resources/objects/furniture/food_bucket.gd [hunk 6]
```diff
@@ -4,9 +4,17 @@ extends Node3D
...
+			var item = item_scene.instantiate()
+			add_child(item)
+			item.global_position = body.position_before_pickup
+	else:
+		body = body as XRToolsPickable
+		if(body and !body.has_left_spawner):
+			body.has_left_spawner = true
+			var item = item_scene.instantiate()
+			add_child(item)
+			item.global_position = body.position_before_pickup
```
Comment (Thyrexx) [inline resources/objects/furniture/food_bucket.gd]: Don't duplicate code. Just delete the line `body = body as Food`.

File: resources/objects/order_paper.tscn
Comment (Thyrexx) [inline resources/objects/order_paper.tscn:1]: Why are ther 2 order_paper.tscn ?
The other is at "resources/order_paper.tscn"

File: resources/order_paper.tscn
Comment (Thyrexx) [inline resources/order_paper.tscn:1]: Why are ther 2 order_paper.tscn ?
The other is at "resources/objects/order_paper.tscn"

File: addons/XRHandtrackingTools/pose_records.tres [hunk 7]
```diff
@@ -994,6 +994,7 @@ bone_poses = {
 }
 
 [resource]
+resource_local_to_scene = true
```
Comment (Thyrexx) [inline addons/XRHandtrackingTools/pose_records.tres:997]: really necessary?

**Review by Thyrexx — COMMENTED** [review]
See comments

File: addons/godot-xr-tools/objects/pickable.gd (hunk 1 above)
Comment (Jakllp) [reply addons/godot-xr-tools/objects/pickable.gd:230]: Unfreezes picked up items. You might be wondering where items are frozen. All the spawner items are. Like the patties, buns, drinks, trays, plates...

File: resources/food/tomato.tscn (hunk 2 above)
Comment (Jakllp) [reply resources/food/tomato.tscn]: Well yes

File: resources/objects/accessoires/tray.tscn (hunk 5 above)
Comment (Jakllp) [reply resources/objects/accessoires/tray.tscn:28]: Yep. Frozen before first pickup - continuous cd seems to be necessary for stuff like it being moved by a patty flipper or other things

File: resources/objects/order_paper.tscn
Comment (Jakllp) [reply resources/objects/order_paper.tscn:1]: That... Shouldn't be there. Huh.

File: resources/order_paper.tscn
Comment (Jakllp) [reply resources/order_paper.tscn:1]: This one shouldn't be there wth

File: addons/XRHandtrackingTools/pose_records.tres (hunk 7 above)
Comment (Jakllp) [reply addons/XRHandtrackingTools/pose_records.tres:997]: Wanted to try it (does it matter? =

File: resources/objects/furniture/food_bucket.gd (hunk 6 above)
Comment (Jakllp) [reply resources/objects/furniture/food_bucket.gd]: Fair

File: addons/godot-xr-tools/objects/pickable.gd (hunk 1 above)
Comment (Thyrexx) [reply addons/godot-xr-tools/objects/pickable.gd:230]: why not use this?
image

File: resources/food/tomato.tscn (hunk 2 above)
Comment (Thyrexx) [reply resources/food/tomato.tscn]: Then it belongs to Food.tscn doesn't it?

File: addons/XRHandtrackingTools/pose_records.tres (hunk 7 above)
Comment (Thyrexx) [reply addons/XRHandtrackingTools/pose_records.tres:997]: Not that much i guess. Just duplicates the resource ergo more memory usage. It doesn't harm anything. (Yet XD)

File: addons/godot-xr-tools/objects/pickable.gd (hunk 1 above)
Comment (Jakllp) [reply addons/godot-xr-tools/objects/pickable.gd:230]: Could be used, yes. Dunno.

Comment (Jakllp) [reply addons/godot-xr-tools/objects/pickable.gd:230]: But that is technically more annoying as this then would need to be set manually for trays etc as well.
And that one line shouldn't do any harm AND technically just defaults everyone to unfrozen (which we do) and you COULD freeze if you needed to.
Don't know what's better. Having that one line or editing all other frozen objects (tray, friesboxes, food (only food.tscn), plates)

Comment (Thyrexx) [reply addons/godot-xr-tools/objects/pickable.gd:230]: okay idc. Hope it won't bite us in the ass

**Pull request merged**